### DISCUSSION TITLE
Setting a default fps and buffering

### DISCUSSION THREAD

**Discussion by jaejinkorean** [NetrisTV/ws-scrcpy#72] [Q&A]
Hi.
Thanks for your work. I'm very happy to use this tool.
I'm using this in remote location. So we would want to start with low fps and buffering. Is there a way to set default fps and buffering value? Is there a configuration that I can set?

**Comment by drauggres** [comment]
Hi. You can change fps "on the fly": click "More" button (three dots), there will be `Change video settings` spoiler. Once applied, new settings will be saved to local storage and next time you will start with them instead of the default settings.

There is no intentional buffering. All frames are decoded and displayed as soon as possible. Outdated frames (if there are any) will be discarded after receiving a whole frame ([I-frame](https://en.wikipedia.org/wiki/Video_compression_picture_types#Intra-coded_(I)_frames/slices_(key_frames))).

You can change bitrate or video size to increase performance. Also you can try to use `H264 Converter` (in v0.6.1 it is called `mse`). It should use hardware acceleration for decoding, if it's available on your platform.

Also you can try beta version from the [`develop`](https://github.com/NetrisTV/ws-scrcpy/tree/develop) branch. It has the `Configure stream` button on the main page: you can change settings before the start.

P.S. you can also change default settings in the code: static field `preferredVideoSettings` in players.
